tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1662,25 +1687,18 @@ ImmutableTextSnapshotRef SwiftEditorDocument::replaceText(
     }
   }
 
-  // Update the old syntax map offsets to account for the replaced range.
-  // Also set the initial AffectedRange to cover any tokens that
-  // the replaced range intersected. This allows for clients that split
-  // multi-line tokens at line boundaries, and ensure all parts of these tokens
-  // will be cleared.
-  Impl.AffectedRange = Impl.SyntaxMap.adjustForReplacement(Offset, Length, Str.size());
-
   return Snapshot;
 }
 
 void SwiftEditorDocument::updateSemaInfo() {
   Impl.AccessMtx.lock();
   auto EditableBuffer = Impl.EditableBuffer;
   auto SemanticInfo = Impl.SemanticInfo;
-  Impl.AccessMtx.unlock(); // Not a recursive mutex, so unlock before processing
+  // We need to release `AccessMtx` before calling into the ASTManager, since it
+  // may call back to the editor for document state.
+  Impl.AccessMtx.unlock();
```

Comment:
FYI this comment change is because I actually hit this lock inversion before but misunderstood it because I looked at Mutex.h from the runtime (non-recursive) instead of Mutex.h from llvm (recursive).